src/ucp/rndv/rndv.c

```diff
@@ -1536,21 +1507,30 @@ static ucs_status_t ucp_rndv_progress_am_zcopy_multi(uct_pending_req_t *self)
 UCS_PROFILE_FUNC_VOID(ucp_rndv_send_frag_put_completion, (self),
                       uct_completion_t *self)
 {
-    ucp_request_t *freq = ucs_container_of(self, ucp_request_t,
-                                           send.state.uct_comp);
-    ucp_request_t *req  = freq->super_req;
+    ucp_request_t *freq  = ucs_container_of(self, ucp_request_t,
+                                            send.state.uct_comp);
+    ucp_request_t *fsreq = freq->super_req;
+    ucp_request_t *sreq  = fsreq->super_req;
 
     /* release memory descriptor */
-    if (freq->send.mdesc) {
-        ucs_mpool_put_inline((void *)freq->send.mdesc);
+    if (freq->send.mdesc != NULL) {
+        ucs_mpool_put_inline((void*)freq->send.mdesc);
     }
 
-    req->send.state.dt.offset += freq->send.length;
-    ucs_assert(req->send.state.dt.offset <= req->send.length);
+    fsreq->send.state.dt.offset += freq->send.length;
+    ucs_assert(fsreq->send.state.dt.offset <= fsreq->send.length);
 
     /* send ATP for last fragment of the rndv request */
-    if (req->send.length == req->send.state.dt.offset) {
-        ucp_rndv_send_frag_atp(req, req->send.rndv_put.rreq_remote_id);
+    if (fsreq->send.length == fsreq->send.state.dt.offset) {
+        ucp_rkey_destroy(fsreq->send.rndv_put.rkey);
+
+        sreq->send.state.dt.offset += fsreq->send.length;
+        ucs_assert(sreq->send.state.dt.offset <= sreq->send.length);
+
+        ucp_rndv_complete_rma_put_zcopy(sreq, 1);
```

Comment:
maybe move ucp_rkey_destroy() from line 1525 inside ucp_rndv_complete_rma_put_zcopy?

Reply:
but `ucp_rkey_destroy()` is done for fragment, not for super request